src/sentry/static/sentry/app/components/charts/utils.tsx

```diff
@@ -128,7 +128,7 @@ export function getSeriesSelection(
   location: Location,
   parameter = 'unselectedSeries'
 ): EChartOption.Legend['selected'] {
-  const unselectedSeries = decodeList(location.query[parameter]) ?? [];
+  const unselectedSeries = decodeList(location?.query[parameter]) ?? [];
```

Comment:
I wonder if we should abstract the `decode(...location...)` pattern a bit? Maybe if we start using hooks we can have functions that just `useContext`.

Reply:
Definitely agree that abstracting the `decode...(location.query...)` pattern is a good idea. Though I'm not sure about hooks as we have a lot of class components using this pattern.

Reply:
I think that we could combine the fallback values into the `decodeX` functions. We frequently coalesce and having simpler return types would help simplify a bunch of code.